fog/overseer/server/src/service.rs

```diff
@@ -118,4 +139,31 @@ where
 
         Ok(response)
     }
+
+    pub fn get_ingest_summaries(&self) -> Result<Json<GetIngestSummariesResponse>, String> {
+        let mut ingest_summaries: Vec<IngestSummary> = Vec::new();
+        for ingest_client in self.ingest_clients.lock().unwrap().iter() {
+            match ingest_client.get_status() {
+                Ok(proto_ingest_summary) => {
+                    log::trace!(
+                        self.logger,
+                        "Ingest summary retrieved: {:?}",
+                        proto_ingest_summary
+                    );
+                    let ingest_summary: IngestSummary = IngestSummary::from(&proto_ingest_summary);
+                    ingest_summaries.push(ingest_summary);
+                }
+                Err(err) => {
+                    let error_message = format!(
+                        "Unable to retrieve ingest summary for node ({}): {}",
+                        ingest_client.get_uri(),
+                        err
+                    );
+                    return Err(error_message);
```

Comment:
I opted to fail fast here instead of returning a partial slate of IngestSummaries. It's cleaner this way, but I understand that it might be useful to have some of the IngestSummaries returned, and we could also attach error messages for those that aren't returned. 

If you have an opinion on this, let me know!

Reply:
I think it would be useful to actually include error per client. If the clients have an identifying URL, maybe the correct response type is something along the lines of (making this up) `Json<HashMap<Uri, JsonResult>>`, where JsonResult is something like `{error: Option<String>, success: Option<IngestSummary>}`

Reply:
Done. Implemented using Result rather than a bespoke struct.